File: app/handlers/status.py
import time
import psutil
from telegram import Update
from telegram.ext import ContextTypes, CommandHandler

# Fungsi untuk mengubah detik menjadi format Jam:Menit:Detik
def get_readable_time(seconds: int) -> str:
    m, s = divmod(seconds, 60)
    h, m = divmod(m, 60)
    d, h = divmod(h, 24)
    
    parts = []
    if d > 0: parts.append(f"{d}d")
    if h > 0: parts.append(f"{h}h")
    if m > 0: parts.append(f"{m}m")
    parts.append(f"{s}s")
    return " ".join(parts)

# Fungsi untuk mengubah bytes ke unit yang paling sesuai (KB, MB, GB)
def get_readable_size(bytes: int) -> str:
    for unit in ['B', 'KB', 'MB', 'GB', 'TB']:
        if bytes < 1024:
            return f"{bytes:.2f} {unit}"
        bytes /= 1024

async def status_handler(update: Update, context: ContextTypes.DEFAULT_TYPE):
    # Hitung uptime (asumsi bot mulai saat script jalan)
    # Jika ingin uptime bot yang presisi, simpan waktu start di variabel global saat main()
    current_time = time.time()
    uptime_seconds = int(current_time - psutil.boot_time()) # Ini uptime Sistem, 
    # Gunakan variabel 'start_time' global jika ingin uptime Proses Bot saja.

    # Statistik Memori & CPU
    process = psutil.Process()
    mem_info = process.memory_info().rss  # Mengambil penggunaan RAM proses ini saja
    cpu_usage = psutil.cpu_percent(interval=0.5)

    text = (
        "<b>📊 Bot System Status</b>\n"
        "----------------------------\n"
        f"<b>⏱ Uptime:</b> <code>{get_readable_time(uptime_seconds)}</code>\n"
        f"<b>🧠 Memory:</b> <code>{get_readable_size(mem_info)}</code>\n"
        f"<b>⚡ CPU:</b> <code>{cpu_usage}%</code>\n"
        "----------------------------"
    )
    await update.message.reply_text(text, parse_mode="HTML")

def register_status_handler(app):
    app.add_handler(CommandHandler("status", status_handler))

File: src/handlers/status.js
import { execSync } from 'child_process';
import process from 'process';

const START_TIME = Date.now();

function readableTime(ms) {
  const s = Math.floor(ms / 1000);
  const d = Math.floor(s / 86400);
  const h = Math.floor((s % 86400) / 3600);
  const m = Math.floor((s % 3600) / 60);
  const sec = s % 60;
  const parts = [];
  if (d) parts.push(`${d}d`);
  if (h) parts.push(`${h}h`);
  if (m) parts.push(`${m}m`);
  parts.push(`${sec}s`);
  return parts.join(' ');
}

function readableSize(bytes) {
  for (const unit of ['B', 'KB', 'MB', 'GB']) {
    if (bytes < 1024) return `${bytes.toFixed(2)} ${unit}`;
    bytes /= 1024;
  }
  return `${bytes.toFixed(2)} TB`;
}

export function registerStatus(bot) {
  bot.command('status', async (ctx) => {
    const uptime = readableTime(Date.now() - START_TIME);
    const memUsage = process.memoryUsage();
    const ram = readableSize(memUsage.rss);
    const cpuUsage = process.cpuUsage();

    const text = `<b>📊 Bot System Status</b>
----------------------------
<b>⏱ Uptime:</b> <code>${uptime}</code>
<b>🧠 Memory:</b> <code>${ram}</code>
<b>🟢 Node.js:</b> <code>${process.version}</code>
----------------------------`;

    await ctx.reply(text, { parse_mode: 'HTML' });
  });
}
